### PR TITLE
Move AdsServiceImpl back to components/brave_ads.

### PR DESCRIPTION
**Pull request by aseren**
Resolves https://github.com/brave/brave-browser/issues/22478

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/conversions/conversion_queue_item_info.h [hunk 1]
```diff
@@ -34,6 +35,9 @@ struct ConversionQueueItemInfo final {
   bool IsValid() const;
 };
 
+std::string ConversionsAsString(
```
[the remark this answers is not in the document]
Comment (aseren) [reply vendor/bat-native-ads/src/bat/ads/internal/conversions/conversion_queue_item_info.h:38]: Oh, sorryan, these are unrelated changed, removed them.

File: browser/brave_ads/ads_service_factory.cc [hunk 2]
```diff
@@ -86,13 +86,17 @@ KeyedService* AdsServiceFactory::BuildServiceInstanceFor(
   auto* history_service = HistoryServiceFactory::GetInstance()->GetForProfile(
       profile, ServiceAccessType::EXPLICIT_ACCESS);
 
-  std::unique_ptr<AdsServiceImpl> ads_service(
-      new AdsServiceImpl(profile,
+  auto* rewards_service =
+      brave_rewards::RewardsServiceFactory::GetInstance()->GetForProfile(
+          profile);
+
+  std::unique_ptr<AdsServiceImpl> ads_service(new AdsServiceImpl(
```
[the remark this answers is not in the document]
Comment (aseren) [reply browser/brave_ads/ads_service_factory.cc]: fixed

File: components/brave_ads/browser/ads_service_impl.cc [hunk 3]
```diff
@@ -41,16 +41,15 @@
...
 #include "brave/browser/brave_browser_process.h"
-#include "brave/browser/brave_rewards/rewards_service_factory.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/common/brave_channel_info.h"
 #include "brave/common/pref_names.h"
 #include "brave/components/brave_adaptive_captcha/buildflags/buildflags.h"
 #include "brave/components/brave_ads/browser/ads_p2a.h"
 #include "brave/components/brave_ads/browser/ads_storage_cleanup.h"
 #include "brave/components/brave_ads/browser/frequency_capping_helper.h"
+#include "brave/components/brave_ads/browser/service_sandbox_type.h"
```
[the remark this answers is not in the document]
Comment (aseren) [reply components/brave_ads/browser/ads_service_impl.cc:53]: It is here: 
https://github.com/brave/brave-core/blob/d809df49dd2e0e8342d25e92185ad459d32295af/components/brave_ads/browser/ads_service_impl.cc#L723

**Review by goodov — APPROVED** [review]

**Review by tmancey — APPROVED** [review]

File: components/brave_ads/browser/BUILD.gn [hunk 4]
```diff
@@ -49,6 +48,42 @@ source_set("browser") {
     sources += [ "ads_tooltips_delegate.h" ]
     deps += [ "//brave/components/brave_adaptive_captcha" ]
   }
+
+  # TODO(https://github.com/brave/brave-browser/issues/17628)
+  check_includes = false
```
Comment (bridiver) [inline components/brave_ads/browser/BUILD.gn:53]: why does this add `check_includes = false`? You added specific include rules below, but why are we moving this to components when it has all of these layering violations? This change doesn't make any sense to me.

**Pull request merged**